### PR TITLE
feat: verifier & signer to fail on invalid RDF

### LINKED ISSUE #440: Verifier & Signer to fail on invalid RDF

### PR DESCRIPTION
**Pull request by sudeshrshetty**
- moved private issuer util code to internal packages
- Closes #440

Signed-off-by: sudesh.shetty <sudesh.shetty@securekey.com>

### CODE REVIEW

File: cmd/vc-rest/go.mod [hunk 1]
```diff
@@ -20,6 +20,4 @@ require (
 
 replace github.com/trustbloc/edge-service => ../..
 
-replace github.com/piprate/json-gold => github.com/trustbloc/json-gold v0.3.1-0.20200414173446-30d742ee949e
```
Comment (troyronda) [inline cmd/vc-rest/go.mod:23]: why is this removed?

File: cmd/vc-rest/go.sum [hunk 2]
```diff
@@ -218,6 +220,10 @@ github.com/stretchr/testify v1.4.0 h1:2E4SXV/wtOkTonXsotYi4li6zVWxYlZuYNCXe9XRJy
 github.com/stretchr/testify v1.4.0/go.mod h1:j7eGeouHqKxXV5pUuKE4zz7dFj8WfuZ+81PSLYec5m4=
 github.com/stretchr/testify v1.5.1 h1:nOGnQDM7FYENwehXlg/kFVnos3rEvtKTjRvOWSzb6H4=
 github.com/stretchr/testify v1.5.1/go.mod h1:5W2xD1RspED5o8YsWQXVCued0rvSQ+mT+I5cxcmMvtA=
+github.com/sudeshrshetty/aries-framework-go v0.0.0-20200514170854-2dd754fc88ba h1:SQF4lKEgt6OIq1yoPvsoRX7nYPcTSDQgR7DzJi9N1SY=
```
Comment (troyronda) [inline cmd/vc-rest/go.sum:223]: what's this?

File: go.sum [hunk 3]
```diff
@@ -212,6 +212,8 @@ github.com/stretchr/testify v1.4.0 h1:2E4SXV/wtOkTonXsotYi4li6zVWxYlZuYNCXe9XRJy
 github.com/stretchr/testify v1.4.0/go.mod h1:j7eGeouHqKxXV5pUuKE4zz7dFj8WfuZ+81PSLYec5m4=
 github.com/stretchr/testify v1.5.1 h1:nOGnQDM7FYENwehXlg/kFVnos3rEvtKTjRvOWSzb6H4=
 github.com/stretchr/testify v1.5.1/go.mod h1:5W2xD1RspED5o8YsWQXVCued0rvSQ+mT+I5cxcmMvtA=
+github.com/sudeshrshetty/aries-framework-go v0.0.0-20200514201803-74667b93daac h1:JemjylmwAAU7FeevCUbPw5MAnUjCE65zOyY92MeiGVo=
```
Comment (troyronda) [inline go.sum:215]: what's this

File: test/bdd/go.sum [hunk 4]
```diff
@@ -260,6 +262,10 @@ github.com/stretchr/testify v1.4.0 h1:2E4SXV/wtOkTonXsotYi4li6zVWxYlZuYNCXe9XRJy
 github.com/stretchr/testify v1.4.0/go.mod h1:j7eGeouHqKxXV5pUuKE4zz7dFj8WfuZ+81PSLYec5m4=
 github.com/stretchr/testify v1.5.1 h1:nOGnQDM7FYENwehXlg/kFVnos3rEvtKTjRvOWSzb6H4=
 github.com/stretchr/testify v1.5.1/go.mod h1:5W2xD1RspED5o8YsWQXVCued0rvSQ+mT+I5cxcmMvtA=
+github.com/sudeshrshetty/aries-framework-go v0.0.0-20200514170854-2dd754fc88ba h1:SQF4lKEgt6OIq1yoPvsoRX7nYPcTSDQgR7DzJi9N1SY=
```
Comment (troyronda) [inline test/bdd/go.sum:265]: what's this

File: cmd/vc-rest/go.mod (hunk 1 above)
Comment (sudeshrshetty) [reply cmd/vc-rest/go.mod]: by mistake, reverted it

File: cmd/vc-rest/go.sum (hunk 2 above)
Comment (sudeshrshetty) [reply cmd/vc-rest/go.sum]: cleaned up

File: go.sum (hunk 3 above)
Comment (sudeshrshetty) [reply go.sum]: cleaned up

File: test/bdd/go.sum (hunk 4 above)
Comment (sudeshrshetty) [reply test/bdd/go.sum]: cleaned up

File: pkg/restapi/issuer/operation/operations.go [hunk 5]
```diff
@@ -794,6 +806,61 @@ func buildCredential(composeCredReq *ComposeCredentialRequest) (*verifiable.Cred
 	return credential, nil
 }
 
+func setContexts(credential *verifiable.Credential, credentialFormatOptions json.RawMessage) error {
```
Comment (fqutishat) [inline pkg/restapi/issuer/operation/operations.go:809]: can we move this function outside operation.go

File: pkg/restapi/issuer/operation/operations.go [hunk 6]
```diff
@@ -794,6 +806,61 @@ func buildCredential(composeCredReq *ComposeCredentialRequest) (*verifiable.Cred
...
+	return err
+}
+
+// decodeContext decodes raw context(s).
+//
+// context can be defined as a single string value or array;
+// at the second case, the array can be a mix of string and object types
+// (objects can express context information); object context are
+// defined at the tail of the array.
+func decodeContext(c interface{}) ([]string, []interface{}, error) {
```
Comment (fqutishat) [inline pkg/restapi/issuer/operation/operations.go:841]: can we move this function outside operation.go

File: pkg/restapi/issuer/operation/operations.go (hunk 5 above)
Comment (sudeshrshetty) [reply pkg/restapi/issuer/operation/operations.go:809]: Sure

File: pkg/restapi/issuer/operation/operations.go (hunk 6 above)
Comment (sudeshrshetty) [reply pkg/restapi/issuer/operation/operations.go:841]: Sure

File: pkg/restapi/issuer/operation/operations.go [hunk 7]
```diff
@@ -19,6 +19,8 @@ import (
 	"strings"
 	"time"
 
+	"github.com/hyperledger/aries-framework-go/pkg/doc/util"
+
```
Comment (rolsonquadras) [inline pkg/restapi/issuer/operation/operations.go:23]: extra line

File: pkg/restapi/issuer/operation/operations.go (hunk 6 above)
Comment (sudeshrshetty) [reply pkg/restapi/issuer/operation/operations.go]: moved this one including other private code from this package to internal.

File: pkg/restapi/issuer/operation/operations.go (hunk 7 above)
Comment (sudeshrshetty) [reply pkg/restapi/issuer/operation/operations.go]: done
